### ISSUE TITLE
Using sbt-server

### ISSUE THREAD

**Issue by laughedelic** [jamesward/atom-build-sbt#2]
Running `sbt compile` separately takes every time at least 20 seconds on my machine. I think this makes this plugin quite useless for me. Would be really cool to integrate it with sbt-server to have incremental compilation and nice client-server interaction without relaunching sbt every time. It might be quite involved, so a simpler solution is to run `sbt ~compile` somehow and parse output.

WDYT?

**Comment by jamesward** [comment]
:+1:

**Comment by inkytonik** [comment]
As @laughedelic has noticed, my [sbt package](https://atom.io/packages/sbt) aims to achieve something similar to the suggestion here. It runs an interactive sbt session but also watches for errors and test failures, notifying the linter package so you can easily jump to the error/failure locations. Version 0.7.0 which I just released properly deals with continuous (tilde) commands. I would welcome any feedback.

**Comment by ches** [comment]
Thanks @inkytonik, I've been trying out your package and appreciate the work on it. In the long run, however, I'd ideally like to have an `atom-build` integration for every language I work with in earnest, where the usage patterns, configuration, key bindings, etc. are consistent and uniformly manageable. It's one of the things I love about Vim, the built-in conventions of the `:make` and `quickfix` system. Your integration with the common `linter` package does indeed help to go a long way in this sense, for navigating errors with common key bindings, etc.

`atom-build` may or may not have ideal support and APIs for something like a client-server model, yet, but it'd be great if we the community could help it to serve such wishes. Plus work to push forward SBT's client-server support can benefit a lot of projects beyond Atom (it seems like a lot of progress has been made largely motivated by Activator, but docs on these features lag woefully behind).

**Comment by inkytonik** [comment]
Yeah @ches, I can totally see that it would be good to have nice sbt integration with the build side of things. I am coming mostly from the SublimeSbt plugin.

It will definitely be interesting to see how the sbt server developments pan out, but I think it's too early to really rely on anything there.
